packages/mask/src/plugins/IdeaMarket/SNSAdaptor/ListingsView.tsx

```diff
@@ -0,0 +1,274 @@
+import { DataGrid, GridColDef, GridRenderCellParams, GridValueFormatterParams } from '@mui/x-data-grid'
+import { formatterToUSD, urlWithoutProtocol } from '../utils'
+import { formatWeiToEther } from '@masknet/web3-shared-evm'
+import { LoadingAnimation } from '@masknet/shared'
+import { useI18N } from '../../../utils/i18n-next-ui'
+import { EMPTY_LIST } from '@masknet/shared-base'
+import { UrlIcon } from '../icons/UrlIcon'
+import { BASE_URL, TWITTER_BASE_URL } from '../constants'
+import { IdeaToken, Markets } from '../types'
+
+const useStyles = makeStyles()((theme) => {
+    return {
+        root: {
+            height: 350,
+        },
+        grid: {
+            '& .MuiDataGrid-row:nth-child(odd)': {
+                backgroundColor: theme.palette.background.default,
+                '&:hover': {
+                    backgroundColor: theme.palette.background.default,
+                },
+            },
+            '& .MuiDataGrid-row:nth-child(even)': {
+                '&:hover': {
+                    backgroundColor: 'inherit',
+                },
+            },
+            '& .MuiDataGrid-cell p': {
+                whiteSpace: 'nowrap',
+                overflow: 'hidden',
+                textOverflow: 'ellipsis',
+            },
+        },
+        market: {
+            color: 'rgba(8,87,224,1)',
+        },
+        empty: {
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+            justifyContent: 'center',
+            height: '100%',
+            padding: theme.spacing(8, 0),
+        },
+        avatar: {
+            marginRight: theme.spacing(0.8),
+            marginLeft: theme.spacing(0.1),
+            width: 27,
+            height: 27,
+        },
+        nameCellContainer: {
+            width: '100%',
+        },
+        url: {
+            marginRight: theme.spacing(0.7),
+            width: 30,
+            height: 30,
+        },
+        profileAvatar: {
+            float: 'left',
+            marginTop: theme.spacing(1.1),
+        },
+    }
+})
+
+function CustomLoading() {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+
+    return (
+        <div className={classes.empty}>
+            <LoadingAnimation />
+        </div>
+    )
+}
+
+function NoRowsCustomOverlay() {
+    const { t } = useI18N()
+
+    return (
+        <Stack height="100%" alignItems="center" justifyContent="center">
+            {t('no_data')}
+        </Stack>
+    )
+}
+
+function NoResultsCustomOverlay() {
+    const { t } = useI18N()
+
+    return (
+        <Stack height="100%" alignItems="center" justifyContent="center">
+            {t('no_data')}
+        </Stack>
+    )
+}
+
+export function ListingsView() {
+    const { classes } = useStyles()
+    const { t } = useI18N()
+    const [searchText, setSearchText] = useState('')
+    const [page, setPage] = useState(0)
+    const [filters, setFilters] = useState(() => ['0x1', '0x6'])
```

Comment:
what is 0x1 and 0x6?

Reply:
It's the link type. 0x1 = Twitter type link, 0x6 = URL type link.
I will modify accordingly.

Reply:
Done.